### ISSUE TITLE
Ktlint should not lint files generated on the fly

### ISSUE THREAD

**Issue by AleksanderBrzozowski** [JLLeitschuh/ktlint-gradle#580]
We have a project in which we use java classes generated from avro files.
In this project, we also use ktlint to lint kotlin code.

When we run build, we see following warning in console:
```shell
Task :runKtlintCheckOverMainSourceSet
Execution optimizations have been disabled for task ':runKtlintCheckOverMainSourceSet' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/Users/xyz/avro-ktlint-gradle/build/generated-main-avro-java'. Reason: Task ':runKtlintCheckOverMainSourceSet' uses this output of task ':generateAvroJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.4.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

In order to get rid of this warning, we need to explicitly set dependency on `generateAvroJava` task, like this:
```groovy
tasks.named("runKtlintCheckOverMainSourceSet").configure { dependsOn(":generateAvroJava") }
```

Problem was also [reported](https://github.com/davidmc24/gradle-avro-plugin/issues/195) to generate avro java plugin.

Is there any way to solve this problem on the plugin side?

**Comment by dalewking** [comment]
Any update on this? This keeps me using the plugin.

**Comment by shiveenp** [comment]
Hello @wakingrufus just calling out that this seems to be a regression with the new JDK/Gradle version and has affected us as well.

**Comment by wakingrufus** [comment]
> Hello @wakingrufus just calling out that this seems to be a regression with the new JDK/Gradle version and has affected us as well.

Hi, are you on the latest version of the plugin? I recently reworked how source sets are detected in a way that may have solved this. If it is still broken, a minimal reproducer would be very helpful to me to dig in further. Thanks.
